commands/sampledata.go

```diff
@@ -0,0 +1,405 @@
+	deactivatedUsers, _ := command.Flags().GetInt("deactivated-users")
+	guests, _ := command.Flags().GetInt("guests")
+	teamMemberships, _ := command.Flags().GetInt("team-memberships")
+	channelMemberships, _ := command.Flags().GetInt("channel-memberships")
+	postsPerChannel, _ := command.Flags().GetInt("posts-per-channel")
+	directChannels, _ := command.Flags().GetInt("direct-channels")
+	postsPerDirectChannel, _ := command.Flags().GetInt("posts-per-direct-channel")
+	groupChannels, _ := command.Flags().GetInt("group-channels")
+	postsPerGroupChannel, _ := command.Flags().GetInt("posts-per-group-channel")
+	profileImagesPath, _ := command.Flags().GetString("profile-images")
+	withAttachments := profileImagesPath != ""
+
+	if teamMemberships > teams {
+		return fmt.Errorf("you can't have more team memberships than teams")
+	}
+	if channelMemberships > channelsPerTeam {
+		return fmt.Errorf("you can't have more channel memberships than channels per team")
+	}
+
+	if users < 6 && groupChannels > 0 {
+		return fmt.Errorf("you can't have group channels generation with less than 6 users. Use --group-channels 0 or increase the number of users")
+	}
+
+	var bulkFile *os.File
+	var tmpDir string
+	var err error
+	switch bulk {
+	case "":
+		tmpDir, err = ioutil.TempDir("", "mmctl-sampledata-")
+		if err != nil {
+			return fmt.Errorf("unable to create temporary directory")
+		}
+		defer os.RemoveAll(tmpDir)
+
+		if withAttachments {
+			if err = os.Mkdir(filepath.Join(tmpDir, attachmentsDir), 0755); err != nil {
+				return fmt.Errorf("cannot create attachments directory: %w", err)
+			}
+		}
+
+		bulkFile, err = os.Create(filepath.Join(tmpDir, "import.jsonl"))
+		if err != nil {
+			return fmt.Errorf("unable to open temporary file: %w", err)
+		}
+		defer bulkFile.Close()
+	case "-":
+		bulkFile = os.Stdout
+	default:
+		bulkFile, err = os.OpenFile(bulk, os.O_RDWR|os.O_CREATE, 0755)
+		if err != nil {
+			return fmt.Errorf("unable to write into the %q file: %w", bulk, err)
+		}
+		defer bulkFile.Close()
+	}
+
+	profileImages := []string{}
+	if profileImagesPath != "" {
+		profileImages, err = processProfileImagesDir(profileImagesPath, tmpDir, bulk)
+		if err != nil {
+			return fmt.Errorf("cannot process profile images directory: %w", err)
+		}
+	}
+
+	encoder := json.NewEncoder(bulkFile)
+	version := 1
+	if err := encoder.Encode(app.LineImportData{Type: "version", Version: &version}); err != nil {
+		return fmt.Errorf("could not encode version line: %w", err)
+	}
+
+	fake.Seed(seed)
+	rand.Seed(seed)
+
+	teamsAndChannels := make(map[string][]string, teams)
+	for i := 0; i < teams; i++ {
+		teamLine := createTeam(i)
+		teamsAndChannels[*teamLine.Team.Name] = []string{}
+		if err := encoder.Encode(teamLine); err != nil {
+			return fmt.Errorf("could not encode team line: %w", err)
+		}
+	}
+
+	teamsList := make([]string, len(teamsAndChannels))
+	for teamName := range teamsAndChannels {
+		teamsList = append(teamsList, teamName)
+	}
```

Comment:
This is probably not going to do what we expect. Since the slice has been pre-allocated, appending to it will add more items on top of the existing (empty) ones.

Reply:
right! :facepalm:

Reply:
Fixed in 5757839